README.md

```diff
@@ -2,4 +2,84 @@
 
 [![Build Status](https://travis-ci.org/SUSE/helm-certgen.svg?branch=master)](https://travis-ci.org/SUSE/helm-certgen)
 
-This is a plugin for [Kubernetes Helm Cli](https://github.com/kubernetes/helm) which enables certificate generation for helm charts.
+This repository is for `certgen` plugin for [Kubernetes Helm](https://github.com/kubernetes/helm) CLI.
+This plugin enables TLS certificate generation for helm charts. 
+
+Helm chart that comes with cert.yaml file, which outlines the requirements around certificates,
+this plugin can be used to generate those certificates and place them as a Kubernetes secret in the appropriate namespace.
+More details about what happens behind the scene can be found @ [how certgen plugin works](docs/how-it-works.md) 
+
+Since the plugin uses Kubernetes certificate API, we suggest that you use it against a Kubernetes version 1.6.4 or above.
+
+If you are using Minikube then please follow the [minikube configuration](docs/minikube-configuration.md) guide to make sure that your minikube is 
+setup correctly for approving CSR and issuing the certificates.
+
+
+## Installing certgen plugin
+There are three steps for installing this plugin successfully.
+
+### 1. Setting up environment
+#### Go runtime
+Currently to use the plugin you need to have **`go`** (1.8) run time installed on your machine so that you can build the plugin binary. To verify that you can run the following command and you should see the expected output
+```
+# go version
+go version go1.8 darwin/amd64
+```
+If you run this on Linux machine, you will see `go version go1.8 linux/amd64`
+
+#### Environment variables
+Make sure GOBIN path is set and is included in your PATH
+```
+export GOBIN=$GOPATH/bin
+export PATH=$PATH:${GOBIN}
+```
+
+### 2. Build plugin binary
+The following command will generate `helm-certgen` binary in GOBIN path
+```
+go get github.com/SUSE/helm-certgen
+```
+
+### 3. Add certgen to helm CLI
+To install the plugin, we will have to copy the wrapper scripts from project's plugin directory to HELM_HOME, so run following commands
+
+```
+HELM_HOME=$(helm home)
+mkdir -p "$HELM_HOME/plugins/certgen"
+cp $GOPATH/src/github.com/SUSE/helm-certgen/plugin/* $HELM_HOME/plugins/certgen/
+```
+
+
+## Using Plugin
+To understand how to use the plugin, we will try to deploy sample/go-demo-app using this plugin and helm cli.
+
+More details on how sample/go-demo-app can be found @ [sample/go-demo-app](sample/go-demo-app/README.md) 
+
+
+### deploying the application
+
+To start with, make sure you have helm API (tiller) enabled on your Kubernetes cluster (using `helm init`).
+Also, make sure you have installed the `certgen` plugin as per [these instructions](#installing-certgen-plugin).
+
+Now let's generate the certificates for this we can run following commands:
+```
+helm certgen generate sample/go-demo-app --namespace sample-go-demo-app
```

Comment:
Before this, we need to cd to the helm-certgen directory.
`cd  $GOPATH/src/github.com/SUSE/helm-certgen/`